rust/otap-dataflow/crates/otap/src/attribute_processor.rs

```diff
@@ -0,0 +1,643 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Attribute processor for OTAP pipelines.
+//!
+//! This processor provides attribute transformations for telemetry data, supporting
+//! the OpenTelemetry Collector's attributes processor configuration format. It operates
+//! on OTAP Arrow payloads (OtapArrowRecords and OtapArrowBytes) and can convert OTLP
+//! bytes to OTAP for processing.
+//!
+//! Currently supports a subset of the collector's operations:
+//! - `update` with `value`: Renames an attribute by changing its key
+//! - `delete`: Removes an attribute by key
+//!
+//! Other operations (`insert`, `upsert`, `hash`, `extract`, `convert`) are accepted in
+//! configuration but not yet implemented.
+//!
+//! Example configuration (YAML):
+//! ```yaml
+//! actions:
+//!   - key: "http.method"
+//!     action: "update"
+//!     value: "rpc.method"    # Renames http.method to rpc.method
+//!   - key: "db.statement"
+//!     action: "delete"       # Removes db.statement attribute
+//! ```
+//!
+//! Implementation uses otel_arrow_rust::otap::transform::transform_attributes for
+//! efficient batch processing of Arrow record batches.
+
+use crate::{OTAP_PROCESSOR_FACTORIES, pdata::OtapPdata};
+use async_trait::async_trait;
+use linkme::distributed_slice;
+use otap_df_config::error::Error as ConfigError;
+use otap_df_config::experimental::SignalType;
+use otap_df_config::node::NodeUserConfig;
+use otap_df_engine::config::ProcessorConfig;
+use otap_df_engine::error::Error as EngineError;
+use otap_df_engine::local::processor as local;
+use otap_df_engine::message::Message;
+use otap_df_engine::processor::ProcessorWrapper;
+use otel_arrow_rust::otap::{
+    OtapArrowRecords,
+    transform::{AttributesTransform, transform_attributes},
+};
+use otel_arrow_rust::proto::opentelemetry::arrow::v1::ArrowPayloadType;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::{BTreeMap, BTreeSet};
+use std::sync::Arc;
+
+/// URN for the AttributeProcessor
+pub const ATTRIBUTE_PROCESSOR_URN: &str = "urn:otap:processor:attribute_processor";
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+/// Actions that can be performed on attributes.
+#[serde(tag = "action", rename_all = "lowercase")]
+pub enum Action {
```

Comment:
This configuration looks better because I think the structure is more like the golang attribute processor, but I'm not sure we want to have all the actions we don't yet support.

I think it probably just makes sense to have:
```rs
    /// Delete an attribute by key.
    Delete {
        /// The attribute key to delete.
        key: String,
    },
    
     Rename {
         // Attribute key to rename.
         key: String,
         
          // What the key will be renamed to.
         value: String
    },
```